### PR TITLE
[WFWIP-273] Update templates for CD18

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/WFWIP-273
Signed-off-by: Ken Wills <kwills@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: templates/eap-cd-image-stream.json [hunk 1]
```diff
@@ -46,6 +46,49 @@
                             "name": "registry.redhat.io/jboss-eap-7-tech-preview/eap-cd-openshift-rhel8:latest"
                         }
                     },
+                    {
+                        "name": "18",
+                        "annotations": {
+                            "description": "JBoss EAP continuous delivery Tech Preview version 18.",
+                            "iconClass": "icon-eap",
+                            "tags": "builder,eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
```
Comment (jmesnil) [inline templates/eap-cd-image-stream.json]: is `java:8` correct? I thought the image was supporting JDK 11?

I have the same question for EAP CD17 image actually as https://access.redhat.com/containers/?tab=tech-details#/registry.access.redhat.com/jboss-eap-7-tech-preview/eap-cd-openshift-rhel8 says:

> JAVA_VERSION=11.0

Comment (luck3y) [reply templates/eap-cd-image-stream.json]: @jmesnil yeah that should be 11, I'll update it and check the others too.

**Comment by luck3y** [comment]
@jmesnil @jfdenise Updated to correct java:8->java:11

File: templates/eap-cd-image-stream.json [hunk 2]
```diff
@@ -306,6 +349,81 @@
...
+                "tags": [
+                    {
+                        "name": "latest",
+                        "annotations": {
+                            "description": "The latest available build of JBoss EAP continuous delivery runtime image [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
+                            "version": "latest",
+                            "openshift.io/display-name": "JBoss EAP continuous delivery (Tech Preview)"
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json:377]: Should runtime image be mentioned there?

File: templates/eap-cd-image-stream.json [hunk 3]
```diff
@@ -306,6 +349,81 @@
...
+                    },
+                    {
+                        "name": "18",
+                        "annotations": {
+                            "description": "JBoss EAP continuous delivery runtime image version 18 [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
+                            "version": "18",
+                            "openshift.io/display-name": "JBoss EAP continuous delivery 18 (Tech Preview)"
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json:395]: Should runtime image be mentioned there?

File: templates/eap-cd-image-stream.json [hunk 4]
```diff
@@ -306,6 +349,81 @@
...
+                    {
+                        "name": "18.0",
+                        "annotations": {
+                            "description": "JBoss EAP continuous delivery 18 runtime image [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
+
+                            "version": "18.0",
+                            "openshift.io/display-name": "JBoss EAP continuous delivery 18 (Tech Preview)"
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json:415]: Should runtime image be mentioned there?

File: templates/eap-cd-image-stream.json [hunk 5]
```diff
@@ -306,6 +349,81 @@
...
+                            "name": "registry.redhat.io/jboss-eap-7-tech-preview/eap-cd-runtime-openshift-rhel8:latest"
+                        }
+                    },
+                    {
+                        "name": "18",
+                        "annotations": {
+                            "description": "JBoss EAP continuous delivery runtime image version 18 [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json]: java:8

File: templates/eap-cd-image-stream.json [hunk 6]
```diff
@@ -306,6 +349,81 @@
...
+            },
+            "spec": {
+                "tags": [
+                    {
+                        "name": "latest",
+                        "annotations": {
+                            "description": "The latest available build of JBoss EAP continuous delivery runtime image [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json]: java:8

File: templates/eap-cd-image-stream.json [hunk 7]
```diff
@@ -306,6 +349,81 @@
...
+
+                        }
+                    },
+                    {
+                        "name": "18.0",
+                        "annotations": {
+                            "description": "JBoss EAP continuous delivery 18 runtime image [Tech Preview]",
+                            "iconClass": "icon-eap",
+                            "tags": "eap,javaee,java,jboss,hidden",
+                            "supports": "eap:7.3,javaee:8,java:8",
```
Comment (jfdenise) [inline templates/eap-cd-image-stream.json]: java:8

**Review by jfdenise — CHANGES_REQUESTED** [review]

File: templates/eap-cd-image-stream.json (hunk 2 above)
Comment (luck3y) [reply templates/eap-cd-image-stream.json:377]: I think we should be careful about changing these at this point, updating java:8->11 is fine, adding runtime, not so sure. In any case they're all marked hidden, so this is something we can update later.

File: templates/eap-cd-image-stream.json (hunk 5 above)
Comment (luck3y) [reply templates/eap-cd-image-stream.json]: Thx, I think I need more coffee :)

**Comment by luck3y** [comment]
Do we want to hold this up for the runtime change? The image is hidden, and not something we'd want users pulling to use for other things anyway, but either way is fine with me.

**Review by jfdenise — APPROVED** [review]

**Pull request merged**